src/Storage/Adapter/RedisCluster.php

```diff
@@ -0,0 +1,315 @@
+<?php
+
+/**
+ * This file is part of the Phalcon Framework.
+ *
+ * (c) Phalcon Team <team@phalcon.io>
+ *
+ * For the full copyright and license information, please view the LICENSE.txt
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Phalcon\Storage\Adapter;
+
+use DateInterval;
+use Exception as BaseException;
+use Phalcon\Storage\Exception as StorageException;
+use Phalcon\Storage\SerializerFactory;
+use Phalcon\Support\Exception as SupportException;
+use RedisCluster as RedisService;
+
+use Throwable;
+
+use function defined;
+use function is_bool;
+use function is_int;
+use function mb_strtolower;
+
+/**
+ * Redis adapter
+ *
+ * @property array $options
+ */
+class RedisCluster extends AbstractAdapter
```

Comment:
It might be best to extend the `Redis` adapter. This way certain methods (such as `decrement` for example) are the same - > less code :)

Reply:
Actually, `select`, `checkConnect`, `checkAuth` and `checkIndex` - that were removed, were private.... yeah, I'll update

Reply:
it tricked me a little, I was producing $options with configured in defaults from Adapter\Redis, but after small refactoring it is ok now.